packages/twenty-server/src/engine/core-modules/calendar/timeline-calendar-event.service.ts

```diff
@@ -1,272 +1,159 @@
       return {
         totalNumberOfCalendarEvents: 0,
         timelineCalendarEvents: [],
       };
     }
 
-    const calendarEventParticipants: TimelineCalendarEventParticipantWithPersonInformation[] =
-      await this.workspaceDataSourceService.executeRawQuery(
-        `SELECT
-            "calendarEventParticipant".*,
-            "person"."nameFirstName" as "personFirstName",
-            "person"."nameLastName" as "personLastName",
-            "person"."avatarUrl" as "personAvatarUrl",
-            "workspaceMember"."nameFirstName" as "workspaceMemberFirstName",
-            "workspaceMember"."nameLastName" as "workspaceMemberLastName",
-            "workspaceMember"."avatarUrl" as "workspaceMemberAvatarUrl"
-        FROM
-            ${dataSourceSchema}."calendarEventParticipant" "calendarEventParticipant"
-        LEFT JOIN
-            ${dataSourceSchema}."person" "person" ON "calendarEventParticipant"."personId" = "person".id
-        LEFT JOIN
-            ${dataSourceSchema}."workspaceMember" "workspaceMember" ON "calendarEventParticipant"."workspaceMemberId" = "workspaceMember".id
-        WHERE
-            "calendarEventParticipant"."calendarEventId" = ANY($1)`,
-        [calendarEvents.map((event) => event.id)],
-        workspaceId,
-      );
-
-    const formattedCalendarEventParticipants: TimelineCalendarEventParticipant[] =
-      calendarEventParticipants.map((participant) => {
-        const firstName =
-          participant.personFirstName ||
-          participant.workspaceMemberFirstName ||
-          '';
-
-        const lastName =
-          participant.personLastName ||
-          participant.workspaceMemberLastName ||
-          '';
-
-        const displayName =
-          firstName || participant.displayName || participant.handle;
+    // We've split the query into two parts, because we want to fetch all the participants without any filtering
+    const [events, total] = await this.calendarEventRepository.findAndCount({
+      where: {
+        id: Any(ids),
+      },
+      relations: {
+        calendarEventParticipants: {
+          person: true,
+          workspaceMember: true,
+        },
+        calendarChannelEventAssociations: {
+          calendarChannel: true,
+        },
+      },
+    });
 
-        const avatarUrl =
-          participant.personAvatarUrl ||
-          participant.workspaceMemberAvatarUrl ||
-          '';
+    // Keep events in the same order as they ids were returned
+    const orderedEvents = events.sort(
+      (a, b) => ids.indexOf(a.id) - ids.indexOf(b.id),
+    );
 
-        return {
-          calendarEventId: participant.calendarEventId,
-          personId: participant.personId,
-          workspaceMemberId: participant.workspaceMemberId,
-          firstName,
-          lastName,
-          displayName,
-          avatarUrl,
+    const timelineCalendarEvents = orderedEvents.map((event) => {
+      const participants = event.calendarEventParticipants.map(
```

Comment:
Maybe we should rewrite the output, and use something that is more like the database entities ?

Reply:
Yes the heavy part is done here instead of the FE but that's usually not how we do it in the rest of the app I think so we could re-arrange that